libmultilabel/linear/preprocessor.py

```diff
@@ -1,216 +1,124 @@
+
+        # fit a collection of raw labels to a binary matrix
+        self.binarizer = MultiLabelBinarizer(classes=dataset.get("classes"), sparse_output=True)
+        if not self.include_test_labels:
+            self.binarizer.fit(dataset["train"]["y"])
+        else:
+            self.binarizer.fit(dataset["train"]["y"] + dataset["test"]["y"])
+        return self
+
+    def transform(self, dataset):
+        """Transform the raw data.
 
         Args:
-            training_data (str | pd.DataFrame, optional): Training data file or dataframe in LibMultiLabel format. Ignored if eval is True. Defaults to None.
-            test_data (str | pd.DataFrame, optional): Test data file or dataframe in LibMultiLabel format. Ignored if test_data doesn't exist. Defaults to None.
-            eval (bool, optional): If True, ignores training data and uses previously loaded state to preprocess test data. Defaults to False.
-            label_file (str, optional): Path to a file holding all labels. Defaults to None.
-            include_test_labels (bool, optional): Whether to include labels in the test dataset. Defaults to False.
-            remove_no_label_data (bool, optional): Whether to remove training instances that have no labels. Defaults to False.
+            dataset : The training and test dataset along with labels with keys 'train', 'test', and labels respectively.
+            The dataset has keys 'x' for input features and 'y' for labels. It also contains 'data_format' to indicate
+            the data format used.
 
         Returns:
-            dict[str, dict[str, sparse.csr_matrix]]: The training and test data, with keys 'train' and 'test' respectively.
-            The data has keys 'x' for input features and 'y' for labels.
+            dict[str, dict[str, sparse.csr_matrix]]: The transformed dataset.
         """
-        if label_file is not None:
-            logging.info(f"Load labels from {label_file}.")
-            with open(label_file, "r") as fp:
-                self.classes = sorted([s.strip() for s in fp.readlines()])
-        else:
-            if test_data is None and include_test_labels:
-                raise ValueError(f"Specified the inclusion of test labels but test file does not exist")
-            self.classes = None
-            self.include_test_labels = include_test_labels
-
-        if self.data_format in {"txt", "dataframe"}:
-            data = self._load_text(training_data, test_data, eval)
-        elif self.data_format == "svm":
-            data = self._load_svm(training_data, test_data, eval)
-
-        if "train" in data:
-            num_labels = data["train"]["y"].getnnz(axis=1)
+        if self.binarizer is None:
+            raise AttributeError("Preprecessor has not been fitted.")
+        ds = copy.deepcopy(dataset)
+        # transform a collection of raw text to a matrix of TF-IDF features
+        if (
+            all(f in {"txt", "dataframe"} for f in (self.data_format, ds["data_format"]))
```

Comment:
User error checks moved earlier.
```suggestion
            if ds["data_format"] in {"txt", "dataframe"}:
```